akka-stream/src/main/scala/akka/stream/Shape.scala

```diff
@@ -45,7 +45,19 @@ sealed abstract class OutPort { self: Outlet[_] ⇒
  * express the internal structural hierarchy of stream topologies).
  */
 object Inlet {
-  def apply[T](toString: String): Inlet[T] = new Inlet[T](toString)
+  /**
+   * Scala API
+   *
+   * Creates a new Inlet with the given name.
+   */
+  def apply[T](name: String): Inlet[T] = new Inlet[T](toString)
```

Comment:
Also, in the scaladoc/javadoc, I think it is important to describe the significance of the name (if any)